pymongo/asynchronous/srv_resolver.py

```diff
@@ -96,8 +96,10 @@ def __init__(
         except Exception:
             raise ConfigurationError(_INVALID_HOST_MSG % (fqdn,)) from None
         self.__slen = len(self.__plist)
-        if self.__slen < 2:
-            raise ConfigurationError(_INVALID_HOST_MSG % (fqdn,))
+        if fqdn == self._resolve_uri(True)[0].target.to_text():
```

Comment:
We can't call resolve() here because we can't do I/O in the constructor.

Reply:
Thanks, fixed in 19d31c79443a3f4350d92d040a53fb782f4775d0